File: binding/src/libqdialog.cpp
#include <QCloseEvent>
#include <QContextMenuEvent>
#include <QDialog>
#include <QEvent>
#include <QKeyEvent>
#include <QObject>
#include <QResizeEvent>
#include <QShowEvent>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QWidget>
#include <qdialog.h>
#include "libqdialog.h"
#include "libqdialog.hxx"

QDialog* QDialog_new(QWidget* parent) {
	 return new VirtualQDialog(parent);
}

QDialog* QDialog_new2() {
	 return new VirtualQDialog();
}

QDialog* QDialog_new3(QWidget* parent, int f) {
	 return new VirtualQDialog(parent, static_cast<QFlags<Qt::WindowType>>(f));
}

libqt_string QDialog_Tr(const char* s) {
	QString _ret = QDialog::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QDialog_Result(const QDialog* self) {
	return self->result();
}

void QDialog_SetVisible(QDialog* self, bool visible) {
	self->setVisible(visible);
}

QSize* QDialog_SizeHint(const QDialog* self) {
	return new QSize(self->sizeHint());
}

QSize* QDialog_MinimumSizeHint(const QDialog* self) {
	return new QSize(self->minimumSizeHint());
}

void QDialog_SetSizeGripEnabled(QDialog* self, bool sizeGripEnabled) {
	self->setSizeGripEnabled(sizeGripEnabled);
}

bool QDialog_IsSizeGripEnabled(const QDialog* self) {
	return self->isSizeGripEnabled();
}

void QDialog_SetModal(QDialog* self, bool modal) {
	self->setModal(modal);
}

void QDialog_SetResult(QDialog* self, int r) {
	self->setResult(r);
}

void QDialog_Finished(QDialog* self, int result) {
	self->finished(result);
}

void QDialog_Connect_Finished(QDialog* self, intptr_t slot) {
    void (*slotFunc)(QDialog*, int) = reinterpret_cast<void (*)(QDialog*, int)>(slot);
    QDialog::connect(self, &QDialog::finished, [self, slotFunc](int result) {
	slotFunc(self, result);
    });
}

void QDialog_Accepted(QDialog* self) {
	self->accepted();
}

void QDialog_Connect_Accepted(QDialog* self, intptr_t slot) {
    void (*slotFunc)(QDialog*) = reinterpret_cast<void (*)(QDialog*)>(slot);
    QDialog::connect(self, &QDialog::accepted, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QDialog_Rejected(QDialog* self) {
	self->rejected();
}

void QDialog_Connect_Rejected(QDialog* self, intptr_t slot) {
    void (*slotFunc)(QDialog*) = reinterpret_cast<void (*)(QDialog*)>(slot);
    QDialog::connect(self, &QDialog::rejected, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QDialog_Open(QDialog* self) {
	self->open();
}

int QDialog_Exec(QDialog* self) {
	return self->exec();
}

void QDialog_Done(QDialog* self, int param1) {
	self->done(param1);
}

void QDialog_Accept(QDialog* self) {
	self->accept();
}

void QDialog_Reject(QDialog* self) {
	self->reject();
}

libqt_string QDialog_Tr2(const char* s, const char* c) {
	QString _ret = QDialog::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QDialog_Tr3(const char* s, const char* c, int n) {
	QString _ret = QDialog::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
void QDialog_QBaseSetVisible(QDialog* self, bool visible) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_SetVisible_IsBase(true);
	vqdialog->setVisible(visible);
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnSetVisible(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_SetVisible_Callback(reinterpret_cast<VirtualQDialog::QDialog_SetVisible_Callback>(slot));
}
}

// Base class handler implementation
QSize* QDialog_QBaseSizeHint(const QDialog* self) {
	auto* vqdialog = dynamic_cast<const VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_SizeHint_IsBase(true);
	return new QSize(vqdialog->sizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnSizeHint(const QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<const VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_SizeHint_Callback(reinterpret_cast<VirtualQDialog::QDialog_SizeHint_Callback>(slot));
}
}

// Base class handler implementation
QSize* QDialog_QBaseMinimumSizeHint(const QDialog* self) {
	auto* vqdialog = dynamic_cast<const VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_MinimumSizeHint_IsBase(true);
	return new QSize(vqdialog->minimumSizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnMinimumSizeHint(const QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<const VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_MinimumSizeHint_Callback(reinterpret_cast<VirtualQDialog::QDialog_MinimumSizeHint_Callback>(slot));
}
}

// Base class handler implementation
void QDialog_QBaseOpen(QDialog* self) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Open_IsBase(true);
	vqdialog->open();
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnOpen(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Open_Callback(reinterpret_cast<VirtualQDialog::QDialog_Open_Callback>(slot));
}
}

// Base class handler implementation
int QDialog_QBaseExec(QDialog* self) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Exec_IsBase(true);
	return vqdialog->exec();
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnExec(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Exec_Callback(reinterpret_cast<VirtualQDialog::QDialog_Exec_Callback>(slot));
}
}

// Base class handler implementation
void QDialog_QBaseDone(QDialog* self, int param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Done_IsBase(true);
	vqdialog->done(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnDone(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Done_Callback(reinterpret_cast<VirtualQDialog::QDialog_Done_Callback>(slot));
}
}

// Base class handler implementation
void QDialog_QBaseAccept(QDialog* self) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Accept_IsBase(true);
	vqdialog->accept();
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnAccept(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Accept_Callback(reinterpret_cast<VirtualQDialog::QDialog_Accept_Callback>(slot));
}
}

// Base class handler implementation
void QDialog_QBaseReject(QDialog* self) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Reject_IsBase(true);
	vqdialog->reject();
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnReject(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_Reject_Callback(reinterpret_cast<VirtualQDialog::QDialog_Reject_Callback>(slot));
}
}

// Derived class handler implementation
void QDialog_KeyPressEvent(QDialog* self, QKeyEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
	vqdialog->keyPressEvent(param1);
	} else {
	self->QDialog::keyPressEvent(param1);
}
}

// Base class handler implementation
void QDialog_QBaseKeyPressEvent(QDialog* self, QKeyEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_KeyPressEvent_IsBase(true);
	vqdialog->keyPressEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnKeyPressEvent(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_KeyPressEvent_Callback(reinterpret_cast<VirtualQDialog::QDialog_KeyPressEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QDialog_CloseEvent(QDialog* self, QCloseEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
	vqdialog->closeEvent(param1);
	} else {
	self->QDialog::closeEvent(param1);
}
}

// Base class handler implementation
void QDialog_QBaseCloseEvent(QDialog* self, QCloseEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_CloseEvent_IsBase(true);
	vqdialog->closeEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnCloseEvent(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_CloseEvent_Callback(reinterpret_cast<VirtualQDialog::QDialog_CloseEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QDialog_ShowEvent(QDialog* self, QShowEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
	vqdialog->showEvent(param1);
	} else {
	self->QDialog::showEvent(param1);
}
}

// Base class handler implementation
void QDialog_QBaseShowEvent(QDialog* self, QShowEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_ShowEvent_IsBase(true);
	vqdialog->showEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnShowEvent(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_ShowEvent_Callback(reinterpret_cast<VirtualQDialog::QDialog_ShowEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QDialog_ResizeEvent(QDialog* self, QResizeEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
	vqdialog->resizeEvent(param1);
	} else {
	self->QDialog::resizeEvent(param1);
}
}

// Base class handler implementation
void QDialog_QBaseResizeEvent(QDialog* self, QResizeEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_ResizeEvent_IsBase(true);
	vqdialog->resizeEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnResizeEvent(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_ResizeEvent_Callback(reinterpret_cast<VirtualQDialog::QDialog_ResizeEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QDialog_ContextMenuEvent(QDialog* self, QContextMenuEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
	vqdialog->contextMenuEvent(param1);
	} else {
	self->QDialog::contextMenuEvent(param1);
}
}

// Base class handler implementation
void QDialog_QBaseContextMenuEvent(QDialog* self, QContextMenuEvent* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_ContextMenuEvent_IsBase(true);
	vqdialog->contextMenuEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnContextMenuEvent(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_ContextMenuEvent_Callback(reinterpret_cast<VirtualQDialog::QDialog_ContextMenuEvent_Callback>(slot));
}
}

// Derived class handler implementation
bool QDialog_EventFilter(QDialog* self, QObject* param1, QEvent* param2) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
	return vqdialog->eventFilter(param1, param2);
	} else {
	return self->QDialog::eventFilter(param1, param2);
}
}

// Base class handler implementation
bool QDialog_QBaseEventFilter(QDialog* self, QObject* param1, QEvent* param2) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_EventFilter_IsBase(true);
	return vqdialog->eventFilter(param1, param2);
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnEventFilter(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_EventFilter_Callback(reinterpret_cast<VirtualQDialog::QDialog_EventFilter_Callback>(slot));
}
}

// Derived class handler implementation
void QDialog_AdjustPosition(QDialog* self, QWidget* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
	vqdialog->adjustPosition(param1);
	} else {
	self->QDialog::adjustPosition(param1);
}
}

// Base class handler implementation
void QDialog_QBaseAdjustPosition(QDialog* self, QWidget* param1) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_AdjustPosition_IsBase(true);
	vqdialog->adjustPosition(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QDialog_OnAdjustPosition(QDialog* self, intptr_t slot) {
	auto* vqdialog = dynamic_cast<VirtualQDialog*>(self);
	if (vqdialog && vqdialog->isVirtualQDialog) {
vqdialog->setQDialog_AdjustPosition_Callback(reinterpret_cast<VirtualQDialog::QDialog_AdjustPosition_Callback>(slot));
}
}

void QDialog_Delete(QDialog* self) {
    delete self;
}

File: binding/include/libqdialog.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qdialog.html
pub const qdialog = struct {

    /// New constructs a new QDialog object.
    pub fn New(parent: ?*anyopaque) QtC.QDialog {
        return qtc.QDialog_new(@ptrCast(parent));
    }


    /// New2 constructs a new QDialog object.
    pub fn New2() QtC.QDialog {
        return qtc.QDialog_new2();
    }


    /// New3 constructs a new QDialog object.
    pub fn New3(parent: ?*anyopaque, f: i64) QtC.QDialog {
        return qtc.QDialog_new3(@ptrCast(parent), @intCast(f));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QDialog_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qdialog.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Result(self: ?*anyopaque, ) i32 {
        return qtc.QDialog_Result(@ptrCast(self));
    }

    pub fn SetVisible(self: ?*anyopaque, visible: bool) void {
        qtc.QDialog_SetVisible(@ptrCast(self), visible);
    }

    /// Allows for overriding the related default method
    pub fn OnSetVisible(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QDialog_OnSetVisible(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetVisible(self: ?*anyopaque, visible: bool) void {
        qtc.QDialog_QBaseSetVisible(@ptrCast(self), visible);
    }

    pub fn SizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QDialog_SizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QDialog_OnSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QDialog_QBaseSizeHint(@ptrCast(self));
    }

    pub fn MinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QDialog_MinimumSizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnMinimumSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QDialog_OnMinimumSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QDialog_QBaseMinimumSizeHint(@ptrCast(self));
    }

    pub fn SetSizeGripEnabled(self: ?*anyopaque, sizeGripEnabled: bool) void {
        qtc.QDialog_SetSizeGripEnabled(@ptrCast(self), sizeGripEnabled);
    }

    pub fn IsSizeGripEnabled(self: ?*anyopaque, ) bool {
        return qtc.QDialog_IsSizeGripEnabled(@ptrCast(self));
    }

    pub fn SetModal(self: ?*anyopaque, modal: bool) void {
        qtc.QDialog_SetModal(@ptrCast(self), modal);
    }

    pub fn SetResult(self: ?*anyopaque, r: i32) void {
        qtc.QDialog_SetResult(@ptrCast(self), r);
    }

    pub fn Finished(self: ?*anyopaque, result: i32) void {
        qtc.QDialog_Finished(@ptrCast(self), result);
    }

    pub fn OnFinished(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QDialog_Connect_Finished(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Accepted(self: ?*anyopaque, ) void {
        qtc.QDialog_Accepted(@ptrCast(self));
    }

    pub fn OnAccepted(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_Connect_Accepted(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Rejected(self: ?*anyopaque, ) void {
        qtc.QDialog_Rejected(@ptrCast(self));
    }

    pub fn OnRejected(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_Connect_Rejected(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Open(self: ?*anyopaque, ) void {
        qtc.QDialog_Open(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnOpen(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_OnOpen(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseOpen(self: ?*anyopaque, ) void {
        qtc.QDialog_QBaseOpen(@ptrCast(self));
    }

    pub fn Exec(self: ?*anyopaque, ) i32 {
        return qtc.QDialog_Exec(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnExec(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QDialog_OnExec(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseExec(self: ?*anyopaque, ) i32 {
        return qtc.QDialog_QBaseExec(@ptrCast(self));
    }

    pub fn Done(self: ?*anyopaque, param1: i32) void {
        qtc.QDialog_Done(@ptrCast(self), param1);
    }

    /// Allows for overriding the related default method
    pub fn OnDone(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QDialog_OnDone(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseDone(self: ?*anyopaque, param1: i32) void {
        qtc.QDialog_QBaseDone(@ptrCast(self), param1);
    }

    pub fn Accept(self: ?*anyopaque, ) void {
        qtc.QDialog_Accept(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnAccept(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_OnAccept(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseAccept(self: ?*anyopaque, ) void {
        qtc.QDialog_QBaseAccept(@ptrCast(self));
    }

    pub fn Reject(self: ?*anyopaque, ) void {
        qtc.QDialog_Reject(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnReject(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_OnReject(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseReject(self: ?*anyopaque, ) void {
        qtc.QDialog_QBaseReject(@ptrCast(self));
    }

    pub fn KeyPressEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_KeyPressEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnKeyPressEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_OnKeyPressEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseKeyPressEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_QBaseKeyPressEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn CloseEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_CloseEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnCloseEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_OnCloseEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCloseEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_QBaseCloseEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn ShowEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_ShowEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnShowEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_OnShowEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseShowEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_QBaseShowEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn ResizeEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_ResizeEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnResizeEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_OnResizeEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseResizeEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_QBaseResizeEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn ContextMenuEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_ContextMenuEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnContextMenuEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_OnContextMenuEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseContextMenuEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_QBaseContextMenuEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn EventFilter(self: ?*anyopaque, param1: ?*anyopaque, param2: ?*anyopaque) bool {
        return qtc.QDialog_EventFilter(@ptrCast(self), @ptrCast(param1), @ptrCast(param2));
    }

    /// Allows for overriding the related default method
    pub fn OnEventFilter(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QDialog_OnEventFilter(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseEventFilter(self: ?*anyopaque, param1: ?*anyopaque, param2: ?*anyopaque) bool {
        return qtc.QDialog_QBaseEventFilter(@ptrCast(self), @ptrCast(param1), @ptrCast(param2));
    }

    pub fn AdjustPosition(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_AdjustPosition(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnAdjustPosition(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QDialog_OnAdjustPosition(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseAdjustPosition(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QDialog_QBaseAdjustPosition(@ptrCast(self), @ptrCast(param1));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QDialog_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qdialog.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QDialog_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qdialog.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QDialog_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/dialog.html#types
pub const enums = struct {
    pub const DialogCode = enum {
        pub const Rejected: i32 = 0;
        pub const Accepted: i32 = 1;
    };

};
